agents/baseAgent.go

```diff
@@ -0,0 +1,63 @@
+package agents
+
+import (
+	"strings"
+
+	"github.com/tmc/langchaingo/chains"
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/tools"
+)
+
+type Agent struct {
+	Llm   llms.LLM
+	Chain chains.Chain
+	Tools []tools.Tool
+}
+
+type AgentInterface interface {
+	Run(input string) string
+}
+
+func FromLlmAndTools(llm llms.LLM, chain chains.Chain, tools []tools.Tool) Agent {
+	return Agent{
+		llm,
+		chain,
+		tools,
+	}
+}
+
+func GetActionAndInput(input string) (action, actionInput string) {
```

Comment:
All exported symbols should have comments.

Separately, this doesn't feel like it should live directly in this package namespace, further, generally, we avoid named returns in Go.